src/client/Microsoft.Identity.Client/ManagedIdentity/V2/Cache/Abstractions/MiCacheKey.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Identity.Client.ManagedIdentity.V2.Cache.Abstractions
+{
+    internal readonly struct MiCacheKey : IEquatable<MiCacheKey>
+    {
+        public Guid TenantId { get; }
+        public Guid ManagedIdentityId { get; }
+        public string TokenType { get; }   // always stored lowercase
+
+        public MiCacheKey(Guid tenantId, Guid managedIdentityId, string tokenType)
+        {
+            TenantId = tenantId;
+            ManagedIdentityId = managedIdentityId;
+            // normalize once
+            TokenType = string.IsNullOrEmpty(tokenType)
+                ? string.Empty
+                : tokenType.Trim().ToLowerInvariant();
+        }
+
+        public static MiCacheKey FromStrings(string tenantId, string managedIdentityId, string tokenType) =>
+            new MiCacheKey(Guid.Parse(tenantId), Guid.Parse(managedIdentityId),
+                string.IsNullOrEmpty(tokenType) ? string.Empty : tokenType.Trim().ToLowerInvariant());
+
+        public bool Equals(MiCacheKey other) =>
+            TenantId.Equals(other.TenantId) &&
+            ManagedIdentityId.Equals(other.ManagedIdentityId) &&
+            StringComparer.OrdinalIgnoreCase.Equals(TokenType, other.TokenType);
+
+        public override bool Equals(object obj) => obj is MiCacheKey k && Equals(k);
+
+        public override int GetHashCode()
```

Comment:
Why do you need to do this? Just for normalizing TokenType? Can't you normalize in the ctor? 

Both struct and record struct will define equality for you.